LibVLCSharp/Shared/Helpers/MarshalUtils.cs

```diff
@@ -98,12 +94,119 @@ internal struct Native
             return resultList.ToArray();
         }
 
+        internal static TU[] Retrieve<T, TU>(IntPtr nativeRef, ArrayOut getRef, Func<IntPtr, T> retrieve,
```

Comment:
Could you please add XML comments on these complex methods: What are each for, how to use them... I'm not sure I'll remember after a while.

Reply:
Done